src/NonBlocking/ConcurrentDictionary/DictionaryImplInt.cs

```diff
@@ -44,7 +44,7 @@ private bool TryClaimSlot(ref int entryKey, int key)
                 }
             }
 
-            return key == entryKeyValue || _keyComparer.Equals(key, entryKey);
+            return key == entryKeyValue || _keyComparer.Equals(key, entryKeyValue);
```

Comment:
Keys that do not contain references are never removed from the table. They are basically write-only. Once a key is installed it only may have or not have a value associated, but the key for the given slot will never be changed. That is just to explain the invariants here and why this was not causing failures. 

Using `entryKey` seems indeed unintentional though - why would we reread the value from the table when we already have it?

Reply:
I see, thanks for the explanation. Seems it is time to stop being lazy and actually watch the original presentation.